client/src/scene/SceneGameArena.ts

```diff
@@ -108,12 +109,28 @@ export class SceneGameArena extends Phaser.Scene {
         if (this.frameTimeElapsed > 1000 / this.FRAMERATE) {
             this.updateUserInput();
             this.updateDrawPlayers();
+            this.updateDrawBoard();
 
             // start next frame
             this.frameTimeElapsed = 0;
         }
     }
 
+    private updateDrawBoard() {
+        for (let row = 15; row < 25; row++) {
+            this.gameState.board[row].forEach((monomino) => {
+                if (monomino) monomino.draw(this);
+            });
+        }
+
+        for (let col = 15; col < 25; col++) {
+            for (let row = 0; row < BOARD_SIZE; row++) {
+                const monomino = this.gameState.board[row][col];
+                if (monomino) monomino.draw(this);
+            }
+        }
```

Comment:
Totally agree. About the algorithm, I'm thinking to implement a lazier drawing mechanism that only redraws lines changed in a clearing.

As for double drawing, it is only the center square region that is redrawn. It can be replaced with a more efficient redraw iteration, but let me poke at improving the laziness first.